mypy/messages.py

```diff
@@ -991,6 +992,171 @@ def untyped_decorated_function(self, typ: Type, context: Context) -> None:
             self.fail('Type of decorated function contains type "Any" ({})'.format(
                 self.format(typ)), context)
 
+    def bad_proto_variance(self, actual: int, tvar_name: str, expected: int,
+                           context: Context) -> None:
+        msg = capitalize("{} type variable '{}' used in protocol where"
+                         " {} one is expected".format(variance_string(actual),
+                                                      tvar_name,
+                                                      variance_string(expected)))
+        self.fail(msg, context)
+
+    def concrete_only_assign(self, typ: Type, context: Context) -> None:
+        self.fail("Can only assign concrete classes to a variable of type '{}'"
+                  .format(self.format(typ)), context)
+
+    def concrete_only_call(self, typ: Type, context: Context) -> None:
+        self.fail("Only concrete class can be given where '{}' is expected"
+                  .format(self.format(typ)), context)
+
+    def note_call(self, subtype: Type, call: Type, context: Context) -> None:
+        self.note("'{}.__call__' has type '{}'".format(strip_quotes(self.format(subtype)),
+                                                       self.format(call)), context)
+
+    def report_protocol_problems(self, subtype: Union[Instance, TupleType], supertype: Instance,
```

Comment:
Report protocol problems _only for implicit_ subtype checks. (Also maybe report protocol problems for structural checks with ``TypedDict``s.)